dff/script/core/message.py

```diff
@@ -50,40 +115,122 @@ class Location(DataModel):
+    """
+
+    phone_number: str
+    first_name: str
+    last_name: Optional[str]
+    dff_attachment_type: Literal["contact"] = "contact"
+
+
+class Invoice(Attachment):
+    """
+    This class is a data model that represents an invoice.
+    It includes title, description, currency name and amount.
+    """
+
+    title: str
+    description: str
+    currency: str
+    amount: int
+    dff_attachment_type: Literal["invoice"] = "invoice"
+
+
+class PollOption(DataModel):
+    """
+    This class is a data model that represents a poll option.
+    It includes the option name and votes number.
+    """
+
+    text: str
+    votes: int = Field(default=0)
+    dff_attachment_type: Literal["poll_option"] = "poll_option"
+
+
+class Poll(Attachment):
+    """
+    This class is a data model that represents a poll.
+    It includes a list of poll options.
+    """
+
+    question: str
+    options: List[PollOption]
+    dff_attachment_type: Literal["poll"] = "poll"
+
+
+class DataAttachment(Attachment):
     """
     This class represents an attachment that can be either
     a file or a URL, along with an optional ID and title.
+    This attachment can also be optionally cached for future use.
     """
 
     source: Optional[Union[HttpUrl, FilePath]] = None
-    id: Optional[str] = None  # id field is made separate to simplify type validation
+    cached_filename: Optional[FilePath] = None
+    id: Optional[str] = None
     title: Optional[str] = None
+    use_cache: bool = True
+
+    async def _cache_attachment(self, data: bytes, directory: Path) -> None:
+        """
+        Cache attachment, save bytes into a file.
+
+        :param data: attachment data bytes.
+        :param directory: cache file where attachment will be saved.
+        """
+
+        title = str(uuid4()) if self.title is None else self.title
+        self.cached_filename = directory / title
+        with open(self.cached_filename, "wb") as file:
+            file.write(data)
+
+    async def get_bytes(self, from_interface: MessengerInterface) -> Optional[bytes]:
+        """
+        Download attachment bytes.
+        If the attachment is represented by URL or saved in a file,
+        it will be downloaded or read automatically.
+        Otherwise, a :py:meth:`~dff.messengers.common.MessengerInterface.populate_attachment`
+        will be used for receiving attachment bytes by ID and title.
+
+        :param from_interface: messenger interface the attachment was received from.
+        """
 
-    def get_bytes(self) -> Optional[bytes]:
-        if self.source is None:
-            return None
         if isinstance(self.source, Path):
             with open(self.source, "rb") as file:
                 return file.read()
-        else:
-            with urlopen(self.source.unicode_string()) as file:
+        elif self.use_cache and self.cached_filename is not None:
+            with open(self.cached_filename, "rb") as file:
```

Comment:
We should check if a file exists (especially for cached files since they can be stored in a temp dir).